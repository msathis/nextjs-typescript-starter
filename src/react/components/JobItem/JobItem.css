.Job {
  margin: 5px 10px 0 5px;
  padding: 5px 10px;
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: space-between;
  color: #313039;
  background: #fff;
}
.JobFlags {
  display: flex;
  align-items: center;
  margin: 5px 0;
}
.Company {
  font-weight: bold;
  padding: 10px 0 5px 0;
}
.Location {
  padding: 0 5px;
}
.Country {
  display: flex;
  color: #9f9baf;
}
.Flag {
  border-radius: 5px;
  background-color: orange;
  display: inline-block;
  padding: 0 5px;
  margin: 0 5px;
  color: #fff;
  font-size: 12px;
}
