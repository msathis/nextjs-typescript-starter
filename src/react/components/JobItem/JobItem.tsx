import React from "react";
import PropTypes from "prop-types";

import "./JobItem.css";

const propTypes = {
  job: PropTypes.shape({
    title: PropTypes.string.isRequired,
    company: PropTypes.shape({
      name: PropTypes.string.isRequired
    }),
    cities: PropTypes.array,
    remotes: PropTypes.array
  }),
  onClick: PropTypes.func,
  style: PropTypes.object
};
const defaultProps = {};

export default function JobItem({ job, style, onClick }) {
  return (
    <div className="Job" style={style} onClick={onClick}>
      <div>
        <div className="Title">{job.title}</div>
        <div className="Company">{job.company.name}</div>
      </div>
      <div className="JobFlags">
        <div className="Country">
          {job.cities.map(city => (
            <div key={city.name} className="Location">
              <span>{city.name}, </span>
              <span>{city.country.name}</span>
            </div>
          ))}
        </div>
        {job.remotes ? <div className="Flag">Remote</div> : null}
      </div>
    </div>
  );
}

JobItem.propTypes = propTypes;
JobItem.defaultProps = defaultProps;
