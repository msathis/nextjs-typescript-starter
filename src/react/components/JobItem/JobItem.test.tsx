import React from "react";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { expect as expectChai } from "chai";
import JobItem from "./JobItem";

configure({ adapter: new Adapter() });

describe("JobItem", () => {
    it("should render component", () => {
        const component = shallow(
            <JobItem
                job={{
                    title: "Test Job",
                    description: "Job Description",
                    company: {
                        name: "Uplearn"
                    },
                    cities: []
                }}
            />
        );

        expect(component.getElements()).toMatchSnapshot();
        expectChai(component.find(".Title")).to.have.lengthOf(1);
        expectChai(component.find(".Title").text()).to.equal("Test Job");
    });
});
