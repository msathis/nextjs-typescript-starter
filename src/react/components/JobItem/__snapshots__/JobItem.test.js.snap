// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobItem should render component 1`] = `
Array [
  <div
    className="Job"
  >
    <div>
      <div
        className="Title"
      >
        Test Job
      </div>
      <div
        className="Company"
      >
        Uplearn
      </div>
    </div>
    <div
      className="JobFlags"
    >
      <div
        className="Country"
      />
    </div>
  </div>,
]
`;
